jsk_topic_tools/src/connection_based_nodelet.cpp

```diff
@@ -52,6 +60,13 @@ namespace jsk_topic_tools
     }
   }
 
+  void ConnectionBasedNodelet::warnNeverSubscribedCallback(const ros::WallTimerEvent& event)
+  {
+    if (!ever_subscribed_) {
+      NODELET_WARN("'%s' subscribes topics only with child subscribers. (`always_subscribe: true` to force it to subscribe)", nodelet::Nodelet::getName().c_str());
```

Comment:
Okay, I will.
You mean onInitPostProcess(), right?

2015年9月29日火曜日、Ryohei Ueda<notifications@github.com>さんは書きました:

> In jsk_topic_tools/src/connection_based_nodelet.cpp
> <https://github.com/jsk-ros-pkg/jsk_common/pull/1140#discussion_r40651338>
> :
>
> > @@ -52,6 +60,13 @@ namespace jsk_topic_tools
> >      }
> >    }
> >
> > +  void ConnectionBasedNodelet::warnNeverSubscribedCallback(const ros::WallTimerEvent& event)
> > +  {
> > +    if (!ever_subscribed_) {
> > +      NODELET_WARN("'%s' subscribes topics only with child subscribers. (`always_subscribe: true` to force it to subscribe)", nodelet::Nodelet::getName().c_str());
>
> always_subscribe requires child class implementation and I didn't
> implemented that features in the most of nodelets.
> Can you remove description about always_subscribe? it may confuse users.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_common/pull/1140/files#r40651338>.
>


-- 
和田 健太郎 / Kentaro Wada
http://wkentaro.com